modules/videoio/include/opencv2/videoio.hpp

```diff
@@ -540,6 +553,18 @@ class CV_EXPORTS_W VideoCapture
      */
     CV_WRAP virtual double get(int propId) const;
 
+    /** @brief Open video file or a capturing device for video capturing
+
+    @param filename name of the opened video file (eg. video.avi) or image sequence (eg.
+    img_%02d.jpg, which will read samples like img_00.jpg, img_01.jpg, img_02.jpg, ...)
+
+    @param apiPreference preferred Capture API to use. Can be used to enforce a specific reader
+    implementation if multiple are available: e.g. CAP_FFMPEG or CAP_IMAGES
+
+    The methods first call VideoCapture::release to close the already opened file or camera.
+     */
+    CV_WRAP virtual bool open(const String& filename, int apiPreference);
```

Comment:
@vpisarev Vtable is changed here. Do we allow changes in the end of vtable (minimal ABI impact)?